ocaml/xapi/repository_helpers.ml

```diff
@@ -467,8 +477,14 @@ module UpdateInfo = struct
       raise Api_errors.(Server_error (invalid_updateinfo_xml, []))
     | ui -> ui
 
-  let of_xml_file xml_file_path =
-    match Xml.parse_file xml_file_path with
+  let assert_no_dup_update_id l =
+    let comp x y = compare x.id y.id in
+    if List.length (List.sort_uniq comp l) = List.length (List.sort comp l) then l
```

Comment:
Something like the code below would be more efficient. However, just keep what you have.

```
let rec has_duplicates = function
| [] -> false
| x::y::_ when x = y -> true
| x::y::z -> has_duplicates (y::z)

let assert_no_dup_update_id l = List.sort l |> has_duplicates |> not
```